src/hotspot/share/opto/compile.cpp

```diff
@@ -2773,6 +2763,17 @@ void Compile::Optimize() {
   // Process inline type nodes now that all inlining is over
   process_inline_types(igvn);
 
+  assert(EnableVectorSupport || !has_vbox_nodes(), "sanity");
+  if (EnableVectorSupport && has_vbox_nodes()) {
+    TracePhase tp("", &timers[_t_vector]);
+    PhaseVector pv(igvn);
+    pv.optimize_vector_boxes();
```

Comment:
`PhaseVector` does perform `PhaseRemoveUseless` (as part of `PhaseVector::do_cleanup()`). 

How does it interact with `remove_root_to_sfpts_edges` happening earlier:
```
  // Now that all inlining is over and no PhaseRemoveUseless will run, cut edge from root to loop
  // safepoints
  remove_root_to_sfpts_edges(igvn);
```

Reply:
Fixed, it was an left behind artifact of a previous check-in, its not moved after PhaseVector completion like in current implementation.

Reply:
Why not handling the vector box optimizations before the `process_inline_types` like before? We can do the safepoint scalarize in `PhaseVector` for vector boxes, including the special handles for vector mask utility. Otherwise, we have to move the vector mask utility handling to `InlineTypeNode`, which looks weird. We can skip the vector boxes in `process_inline_types()`. Another benefit is most of the `VectorBoxNode` are eliminated before processing `InlineTypeNode`.